### PR TITLE
add release upload wrappers using gh cli

### PR DESCRIPTION
**Pull request by mrcaseb**
This pr adds code to upload data to releases using the gh cli as an alternative piggyback

### CODE REVIEW

**Comment by mrcaseb** [comment]
Right now this doesn't change anything. It just adds the low level function to upload data to a release using the gh cli

File: R/gh_cli.R [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                                  tag,
+                                  ...,
+                                  repo = "nflverse/nflverse-data",
+                                  overwrite = TRUE){
+  # see https://cli.github.com/manual/gh_release_upload
+
+  # validate file paths
+  file_available <- file.exists(files)
+  if ( !all(file_available) ){
+    cli::cli_abort("The following file{?s} {?is/are} missing: {.path {files[!file_available]}}")
```
Comment (mrcaseb) [inline R/gh_cli.R:26]: I am not perfectly happy with this behavior. Maybe we should warn about missing files and go on with the code IF there are any valid file names

File: R/gh_cli.R [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# See cli manual at https://cli.github.com/manual/
+
+# This functions tries the gh command in a terminal. If it errors, the gh cli
+# isn't available on the machine or at least not on the PATH variable
+gh_cli_available <- function(){
+  gh_test <- try(system2("gh", stderr = TRUE, stdout = FALSE), silent = TRUE)
+
+  if ( inherits(gh_test, "try-error") ){
+    cli::cli_alert_warning("The gh cli is not available on your machine!")
```
Comment (mrcaseb) [inline R/gh_cli.R:9]: I wonder if we should error here

File: R/gh_cli.R (hunk 1 above)
Comment (mrcaseb) [reply R/gh_cli.R]: DONE

File: R/gh_cli.R (hunk 2 above)
Comment (mrcaseb) [reply R/gh_cli.R]: DONE

File: R/upload.R [hunk 3]
```diff
@@ -2,19 +2,23 @@
...
+  # append timestamp files to the actual files to upload
+  # timestamp will be right BEFORE the upload begins instead of afterwards
+  # but it won't be released at all if the upload breaks
+  files <- c(files, timestamp_files)
+
+  gh_cli_release_upload(files = files, tag = tag, repo = repo, overwrite = overwrite)
 }
 
-update_release_timestamp <- function(tag, repo = "nflverse/nflverse-data"){
+path_to_timestamp_files <- function(){
```
Comment (tanho63) [inline R/upload.R:21]:
```suggestion
create_timestamp_file <- function(){
```

**Comment by mrcaseb** [comment]
I just tried the new command and it failed because the shell input command was too long (lot's of file paths). I need to figure out how to do it instead. Maybe system() with a input file.

**Comment by tanho63** [comment]
could paste without collapsing so you run many gh commands?

**Comment by mrcaseb** [comment]
> could paste without collapsing so you run many gh commands?

That's what I had first but felt like it was inefficient. I'll try to figure out a file input and if it doesn't work I will return to many gh commands

**Review by tanho63 — APPROVED** [review]
LGTM. if the max length thing is still an issue later we can switch to removing the collapse parameter

**Pull request merged**